hack/test/unit

```diff
@@ -12,22 +12,42 @@
 #
 set -eux -o pipefail
 
-BUILDFLAGS=(-tags 'netgo seccomp libdm_no_deferred_remove')
-TESTFLAGS+=" -test.timeout=${TIMEOUT:-5m}"
+# run_test_unit runs unit tests for packages selected.
+# arguments:
+# $1 - prefix to use in generated file names. For example, '-libnetwork'.
+# $2 - packages to test
+run_test_unit() {
+	mkdir -p bundles
+	gotestsum --format=standard-quiet --jsonfile=bundles/go-test-report${1}.json --junitfile=bundles/junit-report${1}.xml -- \
+		-tags 'netgo seccomp libdm_no_deferred_remove' \
+		-cover \
+		-coverprofile=bundles/profile${1}.out \
+		-covermode=atomic \
+		${TESTFLAGS:-} \
+		-test.timeout=${TIMEOUT:-5m} \
+		${2}
+}
+
 TESTDIRS="${TESTDIRS:-./...}"
 exclude_paths='/vendor/|/integration'
 pkg_list=$(go list $TESTDIRS | grep -vE "($exclude_paths)")
 
-echo "${pkg_list}" | grep --fixed-strings "libnetwork/drivers/bridge" \
+# NOTE: libnetwork tests require sequential execution.
+# See https://github.com/moby/moby/issues/42458#issuecomment-873216754 for more info.
+# All packages in `/libnetwork` namespace are extracted into a separate package set to account for that below.
+
+base_pkg_list=$(echo "${pkg_list}" | grep --fixed-strings -v "/libnetwork" || :)
+libnetwork_pkg_list=$(echo "${pkg_list}" | grep --fixed-strings "/libnetwork" || :)
+
+echo "${libnetwork_pkg_list}" | grep --fixed-strings "libnetwork/drivers/bridge" \
 	&& if ! type docker-proxy; then
 		hack/make.sh binary-proxy install-proxy
 	fi
 
-mkdir -p bundles
-gotestsum --format=standard-quiet --jsonfile=bundles/go-test-report.json --junitfile=bundles/junit-report.xml -- \
-	"${BUILDFLAGS[@]}" \
-	-cover \
-	-coverprofile=bundles/profile.out \
-	-covermode=atomic \
-	${TESTFLAGS} \
-	${pkg_list}
+if [ "${base_pkg_list}" ]; then
+	run_test_unit '' "${base_pkg_list}"
+fi
+if [ "${libnetwork_pkg_list}" ]; then
+	# Run only one test binary in parallel to ensure sequential execution.
```

Comment:
Perhaps combine the "NOTE" above with this comment; I think the comments combined makes it clearer what's happening. Something like;

```suggestion
	# libnetwork tests invoke iptables, and cannot be run in parallel. Execute
	# tests within /libnetwork with '-p=1' to run them sequentially. See
	# https://github.com/moby/moby/issues/42458#issuecomment-873216754 for details.
```